### PR TITLE
Deps check

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

File: tests/headers.rs [hunk 1]
```diff
@@ -63,20 +68,37 @@ fn check_file_deps<P: AsRef<Path>>(
...
     files_deps.remove(&filename);
 }
+
 fn read_dirs<P: AsRef<Path>>(
     dir: P,
     files_deps: &mut HashMap<String, Vec<String>>,
     errors: &mut u32,
-    level: u32
-) {
+    level: u32) {
```
Comment (retep998) [inline tests/headers.rs]: Please stick to block indentation.

File: tests/headers.rs [hunk 2]
```diff
@@ -104,13 +127,16 @@ fn check_imports() {
                 break;
             }
             let parts: Vec<&str> = line.split("&[").collect();
-            let deps = parts[1].split(',').map(|x| get_between_quotes(x).to_owned()).collect();
-            files_deps.insert(format!("{}.rs", get_between_quotes(parts[0])), deps);
+            files_deps.insert(format!("{}.rs", get_between_quotes(parts[0])),
+                              parts[1].split(',')
+                                      .map(|x| get_between_quotes(x).to_owned())
+                                      .filter(|x| !x.is_empty())
+                                      .collect());
```
Comment (retep998) [inline tests/headers.rs]: Why don't you like assigning deps to an intermediate variable first? It really helps with minimizing indentation.

File: tests/headers.rs [hunk 3]
```diff
@@ -63,20 +68,37 @@ fn check_file_deps<P: AsRef<Path>>(
...
+                }
+            }
+            if found.len() != entries.len() {
+                for found in found {
+                    if let Some(pos) = entries.iter().position(|x| **x == found) {
+                        entries.remove(pos);
+                    }
+                }
+                if !entries.is_empty() {
+                    io::stderr().write(format!("{}: include not used: {:?}\n",
```
Comment (retep998) [inline tests/headers.rs]: Use `writeln!` here.

File: tests/headers.rs [hunk 4]
```diff
@@ -4,40 +4,44 @@
...
     content
 }
+
 fn check_if_in_build<P: Debug>(
     path: &P,
     include: &str,
     entries: &[String],
-    errors: &mut u32
-) {
+    errors: &mut u32) -> bool {
```
Comment (retep998) [inline tests/headers.rs]: Block indentation!

File: tests/headers.rs [hunk 5]
```diff
@@ -4,40 +4,44 @@
...
+    false
 }
+
 fn check_file_deps<P: AsRef<Path>>(
     p: P,
     files_deps: &mut HashMap<String, Vec<String>>,
     errors: &mut u32,
-    level: u32
-) {
+    level: u32) {
```
Comment (retep998) [inline tests/headers.rs]: Block indentation!

File: tests/headers.rs (hunk 2 above)
Comment (GuillaumeGomez) [inline tests/headers.rs]: It's more straightforward to read.

**Comment by GuillaumeGomez** [comment]
Updated.

**Pull request merged**